### PR TITLE
fix: increase timeout and use tearDownNodes

### PR DESCRIPTION
**Pull request by fbarbu15**
## Problem

There are some tests that consistently fail during the CI at teardown (stopping nodes)

## Solution

Added a bigger timeout and used the tearDownNodes function that better handles this job

### CODE REVIEW

**Comment by fbarbu15** [comment]
Seems that it helps a little but doesn't fix the issue fully. Still 1/3 go waku jobs failed

File: packages/tests/tests/waku.node.spec.ts [hunk 1]
```diff
@@ -63,8 +68,8 @@ describe("Waku Dial [node only]", function () {
     let nwaku: NimGoNode;
 
     afterEach(async function () {
-      !!nwaku && (await nwaku.stop());
-      !!waku && waku.stop().catch((e) => console.log("Waku failed to stop", e));
+      this.timeout(10000);
+      tearDownNodes([nwaku], [waku]);
```
Comment (weboko) [inline packages/tests/tests/waku.node.spec.ts:72]: Is it only this `spec` that is failing with such a reason?

**Comment by fryorcraken** [comment]
Did you check the nwaku logs see if there is any indication of the slow shutting down?

**Comment by fbarbu15** [comment]
> Did you check the nwaku logs see if there is any indication of the slow shutting down?

No, I just saw that the default timeout of [2000 ms is hit](https://github.com/waku-org/js-waku/pull/1592#issuecomment-1730818415) and increasing it to 10000 helps.
However even with 10000 ms there's something wrong happening sometimes, because it fails during the test with [empty payload](https://github.com/waku-org/js-waku/actions/runs/6259161227/job/17011366933?pr=1592)

My guess is that is somehow related to the `stop` of the `AsyncIterator` because this is the only test that uses this function
cc @weboko

**Comment by weboko** [comment]
> My guess is that is somehow related to the stop of the AsyncIterator because this is the only test that uses this function 

Might be a bug in the `AsyncIterator`, need to investigate

**Comment by weboko** [comment]
@danisharora099 https://github.com/waku-org/js-waku/issues/1560

**Review by weboko — APPROVED** [review]

**Comment by fbarbu15** [comment]
> @danisharora099 #1560

**Comment by fbarbu15** [comment]
> > My guess is that is somehow related to the stop of the AsyncIterator because this is the only test that uses this function
> 
> Might be a bug in the `AsyncIterator`, need to investigate

Had a 2nd look and turns out that stop was fired imediately after the message was send and that was the reason for the failures. Adding a small delay seems to fix the issue

**Pull request merged**